.gitlab-ci.yml

```diff
@@ -204,6 +204,21 @@ build_kernel_version_testing_arm64:
     DOCKERFILE: kernel-version-testing/kernel-version-testing_arm64/Dockerfile
     IMAGE: kernel-version-testing_arm64
 
+build_circleci_runner:
+  extends: [ .x64 ]
+  stage: build
+  except: [ tags ]
+  variables:
+    DOCKERFILE: circleci/Dockerfile
+    IMAGE: circleci-runner
+  script:
+    # Dockerhub login
+    - DOCKER_REGISTRY_LOGIN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.docker_hub_login --with-decryption --query "Parameter.Value" --out text)
+    - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.docker_hub_pwd --with-decryption --query "Parameter.Value" --out text | docker login --username "$DOCKER_REGISTRY_LOGIN" --password-stdin docker.io
+    # Build
+    - docker build --build-arg BASE_IMAGE=$BASE_IMAGE --build-arg DD_TARGET_ARCH=$DD_TARGET_ARCH $CUSTOM_BUILD_ARGS --tag datadog/agent-buildimages-$IMAGE:v$CI_PIPELINE_ID-${CI_COMMIT_SHA:0:7}${ECR_TEST_ONLY} --file $DOCKERFILE .
```

Comment:
I have a slight preference for handling this the same way other images are done.

But, as I understand it, this would require creating a new repository in dockerhub for each existing repository, which is probably too much effort for what it's worth. Let's just document the difference in the Go upgrade docs.

Reply:
I've updated the Go upgrade documentation, even if here, I could have just added a second docker repo as we have only one CircleCI image.